testdata/dnn/download_models.py

```diff
@@ -112,6 +116,27 @@ def save(self, r):
                 print('>', end='')
                 sys.stdout.flush()
 
+    def handle_bad_download(self):
+        if os.path.exists(self.filename):
+            # rename file for further investigation
+            try:
+                # NB: using `self.sha_actual` may create unbounded number of files
+                rename_target = self.filename + '.invalid'
+                # TODO: use os.replace (Python 3.3+)
+                try:
+                    if os.path.exists(rename_target):  # avoid FileExistsError on Windows from os.rename()
+                        os.remove(rename_target)
+                finally:
+                    os.rename(self.filename, rename_target)
+                    print('  downloaded content is renamed to ' + rename_target)
```

Comment:
I recommend clearly stating this is an error condition and the action taken. Current wording is vague. For example, a change of
"downloaded content is renamed to"
to
"renaming invalid file to".

That makes the two error conditions express their action
"renaming invalid file to"
"deleting invalid file"

Reply:
Changed